### PR TITLE
The flag "decoded-from-explicit" must be imp/exportable

### LINKED ISSUE #18600: -x509_strict deficiency in FIPS mode
ca.pem
```
-----BEGIN CERTIFICATE-----
MIICbzCCAhagAwIBAgIBATAKBggqhkjOPQQDAjAVMRMwEQYDVQQKDApFeGFtcGxl
IENBMB4XDTE3MDYxNzA5NTUzMFoXDTMyMDYxNzA5NTUzMFowFTETMBEGA1UECgwK
RXhhbXBsZSBDQTCCAUswggEDBgcqhkjOPQIBMIH3AgEBMCwGByqGSM49AQECIQD/
////AAAAAQAAAAAAAAAAAAAAAP///////////////zBbBCD/////AAAAAQAAAAAA
AAAAAAAAAP///////////////AQgWsY12Ko6k+ez671VdpiGvGUdBrDMU7D2O848
PifSYEsDFQDEnTYIhucEk2pmeOETnSa3gZ9+kARBBGsX0fLhLEJH+Lzm5WOkQPJ3
A32BLeszoPShOUXYmMKWT+NC4v4af5uO5+tKfA+eFivOM1drMV7Oy7ZAaDe/UfUC
IQD/////AAAAAP//////////vOb6racXnoTzucrC/GMlUQIBAQNCAATVa8zvVYBe
TnSj4ynwajk5RVvVQGqANR4g0mgRrgc0UDJU+KRnK6Gxi+W2vHkTi8xwqLk8/NQ7
KAZB+CDtZoSro2MwYTAPBgNVHRMBAf8EBTADAQH/MA4GA1UdDwEB/wQEAwIBBjAd
BgNVHQ4EFgQU7KwyVJdtqi7fm0YDe8dznRydWuwwHwYDVR0jBBgwFoAU7KwyVJdt
qi7fm0YDe8dznRydWuwwCgYIKoZIzj0EAwIDRwAwRAIgUvFjxXmResmqseHjKCzQ
W3MK55dQ6mMG184/wQYrnV0CIEm3DInwR6k+EnkuUSVzbXsJSyZ/cpMfMDls/Hso
AwjW
-----END CERTIFICATE-----
```
leaf.pem
```
-----BEGIN CERTIFICATE-----
MIICSTCCAfCgAwIBAgIBAjAKBggqhkjOPQQDAjAVMRMwEQYDVQQKDApFeGFtcGxl
IENBMB4XDTIyMDYxNzA5NTUzMFoXDTIzMDYxNzA5NTUzMFowFDESMBAGA1UEAwwJ
bG9jYWxob3N0MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAryKHE226
C92Gt2OxTyLOtOB7jvdzdfDUHpcuaAnSYpv5I4VBhA3MOLsJEnDReyDzfkUDfyJx
6x3BzrMwrPxU44k8IN7i254g2YWff3J+w/U9nwOxcb2k4GIFqlfRRR2NuzDTEP7i
yaQkxTqnY1d3u+w1e/NZkkNJWzJGXruSxxRaN91zcTvVVy3S8tlQ6/X3JS1GZyN+
qr2bQD6dwlMveOTGH4Ov+sHaPEWe271uORgIdkt5hdgLNw5k0Uy+dhbWVKNguCs+
NYRbG3ftsOfLcoLWmETO/3+S4M1QhV1DpDWviv/5RsVAEFu0GgXNg6YVM/56NMiJ
0EpUYIYkjNvA7QIDAQABo2cwZTAOBgNVHQ8BAf8EBAMCA6gwEwYDVR0lBAwwCgYI
KwYBBQUHAwEwHQYDVR0OBBYEFNEnFhq3dIpxGCNjiu1tDuFChz2cMB8GA1UdIwQY
MBaAFOysMlSXbaou35tGA3vHc50cnVrsMAoGCCqGSM49BAMCA0cAMEQCIFEwt66E
mf0MWjt8NCIGvVMjOmuDSjGenHPsK2PpP/+5AiAm/bKw4Xk/u0xr8/ihovjscdLh
o5vxBl7mOyYLdm/yUQ==
-----END CERTIFICATE-----
```
reproducer:
```
[dbelyavs@localhost openssl]$ ./util/wrap.pl apps/openssl verify -x509_strict -CAfile cacrt.pem leaf.pem
O=Example CA
error 94 at 1 depth lookup: Certificate public key has explicit ECC parameters
error leaf.pem: verification failed
[dbelyavs@localhost openssl]$ ./util/wrap.pl -fips apps/openssl verify -x509_strict -CAfile cacrt.pem leaf.pem
leaf.pem: OK
```

### PR DESCRIPTION
**Pull request by t8m**
Otherwise the information that the EC group was imported from
explicit parameters is lost when the key is moved across providers.

Fixes #18600

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
> @t8m isn't it worth also having a test case from #18600?

Added slightly different but equivalent testcases. (I've verified it reproduces the problem.)

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by beldmit** [comment]
@t8m could you please rebase this PR?

**Comment by t8m** [comment]
Rebased. Unfortunately this also means it won't cherry pick cleanly to 3.0.

**Comment by beldmit** [comment]
The error looks real though unrelated to this PR. Still approved.

File: doc/man7/EVP_PKEY-EC.pod [hunk 1]
```diff
@@ -70,8 +70,8 @@ I<order> multiplied by the I<cofactor> gives the number of points on the curve.
 
 =item  "decoded-from-explicit" (B<OSSL_PKEY_PARAM_EC_DECODED_FROM_EXPLICIT_PARAMS>) <integer>
 
-Gets a flag indicating whether the key or parameters were decoded from explicit
-curve parameters. Set to 1 if so or 0 if a named curve was used.
+Sets or gets a flag indicating whether the key or parameters were decoded from
+explicit curve parameters. Set to 1 if so or 0 if a named curve was used.
```
Comment (mattcaswell) [inline doc/man7/EVP_PKEY-EC.pod:74]: As I've previously mentioned (in an issue/PR somewhere that I forget) our documentation is very unclear about what parameters are gettable, which are settable, which are importable and which are exportable. This documentation now says the flag can be "set" - but in fact it does not appear in the "settables" list. This change actually makes it importable/exportable.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: doc/man7/EVP_PKEY-EC.pod (hunk 1 above)
Comment (t8m) [reply doc/man7/EVP_PKEY-EC.pod:74]: Hmm, right. It is not settable. And IMO should not be. So I am just going to drop this change. It is not my ambition in this PR to fix the documentation of importable parameters.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
@beldmit - still ok after @t8m's latest change?

**Comment by t8m** [comment]
Merged to master branch. Thank you for the reviews.

**Pull request closed without merging**